File: src/main/java/dev/swe573/whatsthis/dto/CommentDto.java
package dev.swe573.whatsthis.dto;

import lombok.Data;

import java.util.ArrayList;
import java.util.List;
import java.util.Date;

@Data
public class CommentDto {

    private Long id;
    private String text;
    private int votes;

    private Long userId;
    private String username;

    private Long postId;

    private Long parentCommentId;
    private List<String> imageUrls = new ArrayList<>();
    
    // New fields for enhanced commenting
    private String commentType; // IDENTIFICATION, ADDITIONAL_INFO, CORRECTION, REQUEST_INFO
    private String confidenceLevel; // MAYBE, LIKELY, VERY_CONFIDENT
    private boolean isExpertOpinion;
    private Date createdAt;
    private List<String> referenceLinks = new ArrayList<>();
}

File: mobile/eurekaNexus/app/src/main/java/com/yusuf/eurekanexus/data/model/CommentDto.kt
package com.yusuf.eurekanexus.data.model

import com.google.gson.annotations.SerializedName

data class CommentDto(
    @SerializedName("text") val text: String,
    @SerializedName("votes") val votes: Int,
    @SerializedName("userId") val userId: Long,
    @SerializedName("username") val username: String,
    @SerializedName("postId") val postId: Long,
    @SerializedName("parentCommentId") val parentCommentId: Long?
)